ccmbioinfo/lfs
# CARDS-2137: Query servlet is too slow;  CARDS-2182: Query servlet sometimes outputs the same node multiple times

Old behavior: for every matching result, load it as a resource, serialize it as a JSON, then ignore all of these JSONs except the few actually needed in the output.
New behavior: from the raw results, only load and serialize the ones needed in the output

To test:
- build with `-Pdocker` (build may fail in the postgres image, in which case run `docker builder prune -f` and build again)
- in `compose-cluster/mssql`: `python3 generate_test_sql.py -n 6000 --basedate 2023-04-11 --time_spread_seconds 0 big_sample.sql` (use the day for "yesterday")
- in `compose-cluster`: `python3 generate_compose_yaml.py --mssql --cards_project cards4prems --oak_filesystem --dev_docker_image --smtps --smtps_test_container --smtps_test_mail_path ~/cardsmail --composum --adminer`
- start with `docker-compose build && docker-compose up`
- import the generated sql file at `http://localhost:1435/`
- import visits at `http://localhost:8080/Subjects.importClarity`, wait for about an hour for the import to finish
- check that manually creating forms from the dashboard is speedy
- check that `http://localhost:8080/query?query=select%20a.form%20from%20[cards:TextAnswer]%20as%20a%20where%20a.value=%27discharged%27&limit=2&offset=1000` is speedy and doesn't contain duplicates
- check that `http://localhost:8080/query?query=select%20c.value,%20d.value%20from%20[cards:DateAnswer]%20as%20d%20inner%20join%20[cards:ResourceAnswer]%20as%20c%20on%20d.form%20=%20c.form%20where%20c.value=%27/Survey/ClinicMapping/-1792626663%27%20and%20d.value%20is%20not%20null%20option(index%20tag%20cards)&limit=2&offset=1000&rawResults=true` returns direct values instead of node JSONs
